smart-contracts/for-eth-devs/index.md

```diff
@@ -0,0 +1,73 @@
+---
+title: Dual Virtual Machine Stack
+description: Overview of Polkadot's dual virtual machine architecture, offering flexibility in choosing the right execution backend for developers.
+categories: Basics, Smart Contracts
+---
+
+# Dual Virtual Machine Stack
+
+## Introduction
+
+Polkadot's smart contract platform provides developers with a unique dual virtual machine architecture, offering flexibility in choosing the right execution backend for their needs. This approach balances immediate Ethereum compatibility with long-term innovation, allowing developers to deploy either unmodified EVM contracts or optimize for higher performance using PVM.
+
+The dual-stack architecture consists of:
+
+- **[REVM Backend](#revm-backend){target=\_blank}**: Provides full Ethereum Virtual Machine compatibility, enabling unchanged Solidity contracts and familiar tooling
+- **[PolkaVM Backend](#polkavm-backend){target=\_blank}**: A RISC-V-based virtual machine optimized for performance and enabling new use cases
+
+Both backends share common infrastructure including RPC interfaces, tooling support, and precompiles, ensuring consistent developer experience while offering distinct performance characteristics.
+
+```mermaid
+graph TD
+    Devs["Developers"] --> REVM["REVM (EVM-compatible)"]
+    Devs --> PVM["PolkaVM (RISC-V)"]
+    REVM --> Infra["Shared Infrastructure<br/>(Ethereum JSON-RPC, Tooling, Precompiles)"]
+    PVM --> Infra
+```
+
+## REVM Backend
+
+The [REVM backend](https://github.com/bluealloy/revm){target=\_blank} integrates a complete Ethereum Virtual Machine implementation, enabling Solidity contracts to run unchanged on Polkadot's smart contract platform.
+
+This allows developers to use their existing Ethereum tooling and infrastructure to build on Polkadot.
+
+The REVM backend is ideal for:
+
+- Quick migration: Projects porting existing Ethereum dApps without code modifications
+- Established tooling: Teams leveraging mature Ethereum development infrastructure
+- Third-party integrations: Applications requiring standard EVM bytecode inspection
+- Compatibility requirements: Contracts needing exact EVM behavior for audited code
```

Comment:
As I was reading REVM I could not help but think: What is the difference between REVM and Ethereum EVM (e.g. Frontier). I think we need to add one sentence that says how REVM is different than Ethereum's EVM. Why didn't Parity just use that. Because as a new dev I want to be sure that I am investing into the right tech and if I am looking for the same as what is on Ethereum, we should say that it is the same yet better.

Reply:
Just a mention that REVM is a... Rust implementation of Ethereum's EVM would suffice.